charts/opentelemetry-collector/templates/configmap-agent.yaml

```diff
@@ -7,5 +7,9 @@ metadata:
     {{- include "opentelemetry-collector.labels" . | nindent 4 }}
 data:
   relay: |
+    {{- if .Values.agentCollector.enabled }}
     {{- include "opentelemetry-collector.agentCollectorConfig" . | nindent 4 -}}
+    {{- else }}
+    {{- include "opentelemetry-collector.baseConfig" . | nindent 4 -}}
+    {{- end }}
```

Comment:
The logic spread in two places now making it hard to read. I would keep these lines untouched and make changes only in `opentelemetry-collector.agentCollectorConfig` instead

Reply:
Ok I will take another look.